py/builtinimport.c

```diff
@@ -133,6 +132,12 @@ STATIC void do_load(mp_obj_t module_obj, vstr_t *file) {
     mp_parse_compile_execute(lex, MP_PARSE_FILE_INPUT, mod_globals, mod_globals);
 }
 
+STATIC void do_load(mp_obj_t module_obj, vstr_t *file) {
+    // create the lexer
+    mp_lexer_t *lex = mp_lexer_new_from_file(vstr_str(file));
+    do_load_from_lexer(module_obj, lex, vstr_str(file));
```

Comment:
Is this doing a proper tail call?  Important to keep stack usage down for imports, since they already use a lot of stack.